app/models/spree/order_inventory.rb

```diff
@@ -98,8 +98,7 @@ def remove_from_shipment(shipment, variant, quantity)
         inventory_unit.destroy
         removed_quantity += 1
       end
-
-      shipment.destroy if shipment.inventory_units.count == 0
+      shipment.destroy if shipment.inventory_units.reload.count == 0
```

Comment:
I know it's not related to this PR at all but... this method has nothing to do with the class it's in (no state or associations used; all method local vars) and it fits perfectly as a `RemoveFromShipment` service

Reply:
Agree on both counts. I'd like to ship the S2 fix asap though... yet another tech debt issue?

Reply:
Not sure it's worth it. I bet it'll stay there forever.

Reply:
Rubocop agrees and it's flagging the complexity. I'd just move on.